### ISSUE TITLE
Hotkeys do not work for other keyboard layouts

### ISSUE THREAD

**Issue by xor111xor** [alacritty/alacritty#7688]
I continue use 0.12.3. On  0.13.1  i don't use hotkeys in other keyboard layout.
Alt+Shift+v  i get paste - ^V
Same problem with Vim i must switch to EN keyboard layout.

### System

OS: arch linux 6.6.15-1-lts
Version: alacritty 0.13.1 (fe2a3c56)
Linux/BSD: X11/Wayland, DE or Compositor and WM: i3-wm, xort

![issue](https://github.com/alacritty/alacritty/assets/129903590/71199c78-b8c2-4692-91d1-014ac961f29c)

**Comment by kchibisov** [comment]
>Same problem with Vim i must switch to EN keyboard layout.

we can't fix it for vim, you should use langmap.

**Comment by xor111xor** [comment]
It's strange that it works on st, rxvt and this old version. FINE)

**Comment by kchibisov** [comment]
Which bindings are you talking about `h`,`j`,`k`,`l` certainly can't work with non latin layout.

**Comment by xor111xor** [comment]
Let's fix the problem with a origin terminal. And maybe the problem with neovim will be fixed (I use the vim-plugin-ruscmd for keymaping)

**Comment by kchibisov** [comment]
idk, for ru you always needed layout mapping, it never worked without it in the first place.

**Comment by xor111xor** [comment]
https://github.com/powerman/vim-plugin-ruscmd/blob/master/plugin/ruscmd.vim
This plugin only clear keymap.
I don't tried reconfig nvim becase same problem in zsh.

**Comment by kchibisov** [comment]
I don't understand what exactly is not working for you. When you type in `ru` you get the `ru` keys, applications are not aware of them, unless you press something like e.g. `ctrl/alt` etc, but it never should work in the first place, because no terminal spec forces you to emit the `latin` chars from the ru layout when you press alt, etc.

If you want so, add bindings.

**Comment by xor111xor** [comment]
When I press Ctrl+Shift+V on another keyboard layout, instead of pasting I get the character ^V printed

**Comment by kchibisov** [comment]
that's expected, ru layout before worked by luck, it never worked in the first place correctly because we don't do latin keysym resolution.

**Comment by xor111xor** [comment]
Luckily, it works on all other terminals. Thank you

**Comment by kchibisov** [comment]
yes, because they do latin keysym  resolution, but alacritty never did that in the first place. So I'm not sure how it worked before correctly. It never worked on e.g. Wayland for us, because we again, just don't do it yet.

**Comment by xor111xor** [comment]
The terminal was normal, I used it for 3 years. I didn't notice any problems with the language.
I haven’t encountered such problems in other terminals for another 10 years))

**Comment by kchibisov** [comment]
it worked only on x11 because latin keys were always used, in any other configuration it always required to setup bindings for non-latin.

**Comment by xor111xor** [comment]
Still, our ancestors were smarter. The universal keyboard is high technology.

**Comment by kchibisov** [comment]
The algorithm to do that is fairly simple on linux, but if you try to do the same on macOS you'll face the fact that you should hard code layouts information somehow to do translation from `ru` key to any other key.

And again you can just bind a bunch of keys if you want `ru` layout working, for me personally it was never a big issue, but I don't use ru layout in my terminal.

**Comment by PRESFIL** [comment]
I confirm that I noticed this bug when I started using a Wayland session in KDE.
Before that I used X-org and didn't notice anything like that.

Also, this is not a `vim` problem, the usual text insertion via <kbd>Control</kbd>+<kbd>Shift</kbd>+<kbd>V</kbd>
in plain `bash` doesn't work on non-english layouts.

**Comment by mahkoh** [comment]
> yes, because they do latin keysym resolution, but alacritty never did that in the first place. So I'm not sure how it worked before correctly. It never worked on e.g. Wayland for us, because we again, just don't do it yet.

Xkb has a little known feature that I've called ctrl fallback: https://docs.rs/kbvm/latest/kbvm/lookup/struct.LookupTable.html#ctrl-transformation-fallback

This feature ad-hoc switches the group from ru to us during the lookup process such that shortcuts such as ctrl-v just work on any layout. If it no longer works then that sounds like you're using _without_modifiers which disables the ctrl modifier and therefore also ctrl fallback.

**Comment by p4vook** [comment]
Is this a wontfix?

**Comment by ealdoshkin** [comment]
> Is this a wontfix? Quite an annoyance tbh.

Last version I checked was 0.14.1, same error.
Still using 0.12.3.
But I switched from Arch to Gentoo.

**Comment by meow-sokovykh** [comment]
Same issue can be reproduced on Mac OS 15.4.1 with alacritty 0.15.1 :(

**Comment by kchibisov** [comment]
>Is this a wontfix? Quite an annoyance tbh.

The issue is not wontfix, just no API from upstream to resolve it. You can generally bind keys in other layouts as well so they'll work or bind by physical code, so it'll work based on key position.

It's not like you can not solve it with just config.

**Comment by p4vook** [comment]
> > Is this a wontfix? Quite an annoyance tbh.
> 
> The issue is not wontfix, just no API from upstream to resolve it. You can generally bind keys in other layouts as well so they'll work or bind by physical code, so it'll work based on key position.
> 
> It's not like you can not solve it with just config.

Perhaps, letting the user choose between physical code and the actual letters would be a reasonable solution?

I'm okay with leaving the default as it is, just having to config every keybind manually is practically unusable.

I suppose, it shouldn't be hard to get the key code corresponding to the standard layout letter, so it can be just one simple bool in a config. What do you think?

**Comment by Yunir** [comment]
For those who struggle with cmd+c and cmd+v on macos russian layout, here is the fix to do in `alacritty.toml`:
```toml
[[keyboard.bindings]]
action = "Copy"
key = "С"    # cyrillic letter
mods = "Command"

[[keyboard.bindings]]
action = "Paste"
key = "М"    # cyrillic letter
mods = "Command"
```
